tests/common/devices/multi_asic.py

```diff
@@ -598,7 +600,7 @@ def check_bgp_session_state_all_asics(self, bgp_neighbors, state="established"):
         """
         for asic in self.asics:
             if asic.namespace in bgp_neighbors:
-                neigh_ips = [k.lower() for k, v in list(bgp_neighbors[asic.namespace].items()) if v["state"] == state]
+                neigh_ips = [k.lower() for k, v in list(bgp_neighbors[asic.namespace].items())]
```

Comment:
The logic here was a bit confusing here. Just like the logic in the `check_bgp_session_state()` method above, we should actually let `neigh_ips` be all the BGP neighbors we want to check, **regardless of their state**, and then call `asic.check_bgp_session_state(neigh_ips, state)` to check if all the neighbors in `neigh_ips` are target state. Otherwise, `asic.check_bgp_session_state(neigh_ips, state)` will always return True because we had already filtered `neigh_ips`.